### ISSUE TITLE
Prepare the release as part of code freeze

### ISSUE THREAD

**Issue by npepinpe** [camunda/camunda#10542]
**Description**

To avoid issues like we had with the RC surprise, we could run the prepare-release goal as part of the code freeze. That way, we would already have all the changes that will go into a release in the release branch _before_ we release, allowing us to test this more easily. The only code changes/commits that would be part of the actual release pipeline would be the ones which prepare the next release version.

**Comment by megglos** [comment]
Would worth doing when migrating the release to GHA.
@megglos look for an issue to migrate the release to GHA and let this depend on that one.

**Comment by Zelldon** [comment]
@megglos is this resolved with the dry runs?

**Comment by megglos** [comment]
As we are performing a full dry release from main every night I would argue we are less likely to run into release issues going forward. What I'm not sure about is what the exact RC surprise was this issue originated from, @npepinpe can you recall?

**Comment by npepinpe** [comment]
That's a good question, I should have written it in the issue. IIRC it was that Atomix had some bug based on the version you pass, I.e. it validated the version at startup and would fail. So we tested a branch with version 8.x.y-whatever which is valid, but then we built something with version 8.x.y-rc, and atomix wanted rc1, or rc2, etc., with a digit. 

That is fixed, but the general idea is still there, I.e. we don't test exactly the release build, but something very close to it it would be good to test exactly what is released.
